ksqldb-parser/src/main/java/io/confluent/ksql/parser/DefaultKsqlParser.java

```diff
@@ -130,4 +142,14 @@ private static String getStatementString(final SingleStatementContext singleStat
         singleStatementContext.stop.getStopIndex()
     ));
   }
+
+  //checks if the error is a reserved keyword error by checking the message and offendingSymbol
+  private static boolean isKeywordError(final String message, final Object offendingSymbol) {
+    final String tokenName = ((CommonToken) offendingSymbol).getText().toLowerCase();
```

Comment:
we should have an `instanceOf` check here - are we sure that for all exceptions the `offendingSymbol` is a `CommonToken`? What happens if it's not a `CommonToken`?

Also we should use `Token` instead of `CommonToken` as it's more resilient to underlying implementation changes

Reply:
if it is not a `Token` then we should just return `false` as if the underlying error happens to be a `reservedKeywordError` then it has to be a `Token` so we would just default to the default error handling procedure if that isn't the case